### PR TITLE
Adapt data model and add book service

### LINKED ISSUE #9: Retrieve book information from third party service
:rocket: Libraary tries to automate as much as possible. That is why we want to get book information by ISBN (where possible).
The backend should implement a new endpoint in the books API:
```http
GET /api/book/{ISBN}
```
This should return a book entity with information loaded from third party providers.
Possible providers would be:
- [Open Library](https://openlibrary.org/)
- [Google books API](https://developers.google.com/books)
- [ISBNdb](https://isbndb.com/)

### PR DESCRIPTION
**Pull request by garlontas**
Closes #9

### CODE REVIEW

File: src/main/java/org/pickwicksoft/libraary/web/rest/AuthorResource.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        log.debug("REST request to delete Author : {}", id);
+        authorRepository.deleteById(id);
+        return ResponseEntity
+            .noContent()
+            .headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, id.toString()))
+            .build();
+    }
+
+    @PostMapping("/book/{bookId}/authors")
+    public ResponseEntity<Author> addAuthorToBook(@PathVariable Long bookId, @RequestBody Author authorRequest) {
```
Comment (ngarlonta) [inline src/main/java/org/pickwicksoft/libraary/web/rest/AuthorResource.java:89]: is this method needed?

File: src/main/java/org/pickwicksoft/libraary/web/rest/BookResource.java [hunk 2]
```diff
@@ -22,21 +26,22 @@
...
-
-    public BookResource(BookRepository bookRepository) {
+    public BookResource(BookRepository bookRepository, BookService bookService) {
         this.bookRepository = bookRepository;
+        this.bookService = bookService;
     }
 
     /**
      * {@code GET  /book/items} : get all the book/items.
+     *
```
Comment (ngarlonta) [inline src/main/java/org/pickwicksoft/libraary/web/rest/BookResource.java:44]: please return a dto

**Review by ngarlonta — CHANGES_REQUESTED** [review]
Fix iss

File: src/main/java/org/pickwicksoft/libraary/domain/Book.java [hunk 3]
```diff
@@ -67,12 +83,21 @@ public void setSubject(String subject) {
         this.subject = subject;
     }
 
-    public String getAuthor() {
-        return author;
+    public void addAuthor(Author author) {
```
[the remark this answers is not in the document]
Comment (garlontas) [reply src/main/java/org/pickwicksoft/libraary/domain/Book.java:86]: Used in controller to add further authors to one book

File: src/main/java/org/pickwicksoft/libraary/domain/Book.java [hunk 4]
```diff
@@ -67,12 +83,21 @@ public void setSubject(String subject) {
...
+        this.authors.add(author);
+        author.getBooks().add(this);
     }
 
-    public void setAuthor(String author) {
-        this.author = author;
+    public void removeAuthor(long authorId) {
```
[the remark this answers is not in the document]
Comment (garlontas) [reply src/main/java/org/pickwicksoft/libraary/domain/Book.java:91]: Same here

File: src/main/java/org/pickwicksoft/libraary/repository/AuthorRepository.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package org.pickwicksoft.libraary.repository;
+
+import java.util.List;
+import org.pickwicksoft.libraary.domain.Author;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface AuthorRepository extends JpaRepository<Author, Long> {
```
[the remark this answers is not in the document]
Comment (garlontas) [reply src/main/java/org/pickwicksoft/libraary/repository/AuthorRepository.java:7]: Does not work like that → it is automated

File: src/main/java/org/pickwicksoft/libraary/web/rest/AuthorResource.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+                if (authorRequest.getId() != null) {
+                    Author _author = authorRepository
+                        .findById(authorRequest.getId())
+                        .orElseThrow(() -> new BadRequestAlertException("Invalid id", ENTITY_NAME, "idinvalid"));
+                    book.addAuthor(_author);
+                    bookRepository.save(book);
+                    return _author;
+                }
+
+                var _author = authorRepository.save(authorRequest);
```
[the remark this answers is not in the document]
Comment (garlontas) [reply src/main/java/org/pickwicksoft/libraary/web/rest/AuthorResource.java:112]: #28

File: src/main/java/org/pickwicksoft/libraary/web/rest/AuthorResource.java (hunk 1 above)
Comment (garlontas) [reply src/main/java/org/pickwicksoft/libraary/web/rest/AuthorResource.java:99]: Yes

File: src/main/java/org/pickwicksoft/libraary/web/rest/AuthorResource.java [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+                book.addAuthor(_author);
+                bookRepository.save(book);
+                return _author;
+            })
+            .orElseThrow(() -> new BadRequestAlertException("Invalid id", ENTITY_NAME, "idinvalid"));
+        return new ResponseEntity<>(author, HttpStatus.CREATED);
+    }
+
+    @DeleteMapping("/book/{bookId}/authors/{authorId}")
+    public ResponseEntity<HttpStatus> deleteAuthorFromBook(
```
[the remark this answers is not in the document]
Comment (garlontas) [reply src/main/java/org/pickwicksoft/libraary/web/rest/AuthorResource.java:122]: Yes

File: src/main/java/org/pickwicksoft/libraary/web/rest/BookResource.java (hunk 2 above)
Comment (garlontas) [reply src/main/java/org/pickwicksoft/libraary/web/rest/BookResource.java:44]: #28

**Review by ngarlonta — APPROVED** [review]

**Pull request merged**